### PR TITLE
Enhanced Ability to modify option for each request

### PR DESCRIPTION
**Pull request by ReiiYuki**
Sometime each request require different options e.g. `content-type`

In this PR, I add options to each request method. It will combine two options together.

### CODE REVIEW

File: lib/RequestFactory.js [hunk 1]
```diff
@@ -10,31 +11,40 @@ class RequestFactory {
...
-    let options = this.getOptions(path, `put`, body)
+  put(path, body, customOptions) {
+    const defaultOptions = this.getOptions(path, `put`, body)
+    const options = this.combineOption(customOptions || {}, defaultOptions)
     return doRequest.call(this, options)
   }
 
-  patch(path, body) {
-    let options = this.getOptions(path, `patch`, body)
+  patch(path, body,customOptions) {
```
Comment (tanapoln) [inline lib/RequestFactory.js]: please format code here

**Comment by tanapoln** [comment]
lgtm with comment

**Comment by tanapoln** [comment]
and please add a new test

**Comment by ReiiYuki** [comment]
Done !

**Review by tanapoln — APPROVED** [review]

**Pull request merged**